src/base/bittorrent/session.cpp

```diff
@@ -2172,6 +2172,10 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     m_addingTorrents.insert(hash, params);
     // Adding torrent to BitTorrent session
     m_nativeSession->async_add_torrent(p);
+    if (params.skipChecking) {
+        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
```

Comment:
Torrent is adding asynchronously so it's very unlikely it is already added at this point.

Reply:
I tested locally and it worked fine for me. Should I put this some other place?

Reply:
I tried something...please check. Don't know much c++